### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by BorzdeG**
0) update Gradle Wrapper
1) refactoring
2) Downloading of javadoc and sources for libraries in Jetbrains IDE is turned off

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,6 +15,18 @@ repositories {
   jcenter()
 }
 
+idea {
```
Comment (ajoberstar) [inline build.gradle:18]: Why wouldn't you want the source and javadoc available in your projects?

File: build.gradle [hunk 2]
```diff
@@ -14,6 +15,18 @@ repositories {
...
+  module {
+    downloadJavadoc = false
+    downloadSources = false
+  }
+}
+
+tasks.withType(AbstractCompile).all {
+  sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (ajoberstar) [inline build.gradle]: Generally I would lean towards setting this as `sourceCompatibility = '1.8'` at the project level rather than on each task.

**Review by ajoberstar — CHANGES_REQUESTED** [review]
Hi @BorzdeG, thanks for contributing! I'm not sure I follow the rationale for the a couple of the changes if you wouldn't mind taking a look at my comments.

File: build.gradle (hunk 1 above)
Comment (BorzdeG) [reply build.gradle:18]: No, I just do not want, chyotba they were downloaded always, but not when it is necessary to the developer - it saves a traffic and time of project initiation. If there is a need, then in IDEA it is always possible to finish loading sourses or javadoc: https://youtrack.jetbrains.com/issue/IDEA-174225

File: build.gradle (hunk 2 above)
Comment (BorzdeG) [reply build.gradle]: I considered the comment. I looked at source codes (the http://bit.ly/2hnsFkX and http://bit.ly/2hoI9VP) and I agree with the comment - it is possible to register compatibility at the level of the project.

File: build.gradle (hunk 1 above)
Comment (ajoberstar) [reply build.gradle:18]: This seems like more of a developer preference thing than something specific to this project. I'd suggest leveraging [Gradle init scripts](https://docs.gradle.org/current/userguide/init_scripts.html) to disable the sources/javadoc for all projects on your machine.

Once you pull this piece out of the PR, I can merge it in.

**Pull request closed without merging**